### ISSUE TITLE
Traefik seems to watch too many rules files with file based config and watch=true

### ISSUE THREAD

**Issue by malc0mn** [containous/traefik#1892]
### Do you want to request a *feature* or report a *bug*?

Possible bug, or undocumented feature as I was unable to find anything related to this case...

### What did you do?
Setup an extremely basic traefik instance with file based and _watched_ config.
```toml
# Some defaults, stripped...

traefikLogsFile = "/var/log/traefik/traefik.log"
accessLogsFile = "/var/log/traefik/access.log"

# The usual entry points, stripped...

[file]
filename = "/etc/traefik/rules.toml"
watch = true
```
Make a backup of `/etc/traefik/rules.toml`
```sh
$ cd /etc/traefik/
$ cp rules.toml rules.toml-bak.20170724-132846
```
Check the logs:
```sh
$ cat /var/log/traefik/traefik.log

time="2017-07-24T15:30:02+02:00" level=debug msg="File event:\"/etc/traefik/rules.toml-bak.20170724-132846\": CREATE" 
time="2017-07-24T15:30:02+02:00" level=debug msg="File event:\"/etc/traefik/rules.toml-bak.20170724-132846\": WRITE" 
```

### What did you expect to see?
Nothing at all in the logs as the file watched should be the one as set in the config...

### What did you see instead?
It appears that traefik is watching any file in /etc/traefik that starts with `rules.toml`? From the logs:
```sh
time="2017-07-24T15:30:02+02:00" level=debug msg="File event:\"/etc/traefik/rules.toml-bak.20170724-132846\": CREATE" 
time="2017-07-24T15:30:02+02:00" level=debug msg="File event:\"/etc/traefik/rules.toml-bak.20170724-132846\": WRITE" 
```

### Output of `traefik version`:

```
Version:      v1.1.2
Codename:     camembert
Go version:   go1.7.4
Built:        2016-12-15_10:27:40AM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Ubuntu 16.04, TOML config:

```toml
graceTimeOut = 10

debug = true

checkNewVersion = false

traefikLogsFile = "/var/log/traefik/traefik.log"
accessLogsFile = "/var/log/traefik/access.log"
logLevel = "ERROR"

ProvidersThrottleDuration = 5
MaxIdleConnsPerHost = 200
InsecureSkipVerify = true

defaultEntryPoints = ["http", "https"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
  # The containers handle this!
  compress = false

[healthcheck]
interval = 30

[web]
address = ":1936"
#CertFile = "traefik.crt"
#KeyFile = "traefik.key"
ReadOnly = true
[web.statistics]
  RecentErrors = 10
[web.auth.basic]
  users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/", "test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]
#  usersFile = "/path/to/.htpasswd"

[file]
filename = "/etc/traefik/rules.toml"
watch = true
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
time="2017-07-24T15:30:02+02:00" level=debug msg="File event:\"/etc/traefik/rules.toml-bak.20170724-132846\": CREATE" 
time="2017-07-24T15:30:02+02:00" level=debug msg="File event:\"/etc/traefik/rules.toml-bak.20170724-132846\": WRITE" 
```

**Comment by malc0mn** [comment]
In `provider/file/file.go` I can see this here:

```go
// Provider holds configurations of the provider.
type Provider struct {
	provider.BaseProvider `mapstructure:",squash"`
	Directory             string `description:"Load configuration from one or more .toml files in a directory"`
}

// Provide allows the file provider to provide configurations to traefik
// using the given configuration channel.
func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *safe.Pool, constraints types.Constraints) error {
	configuration, err := p.loadConfig()

	if err != nil {
		return err
	}

	if p.Watch {
		var watchItem string

		if p.Directory != "" {
			watchItem = p.Directory
		} else {
			watchItem = p.Filename
		}

		if err := p.addWatcher(pool, watchItem, configurationChan, p.watcherCallback); err != nil {
			return err
		}
	}

	sendConfigToChannel(configurationChan, configuration)
	return nil
}
```

But I have setup a file, not a directory... So is this some kind of default that can be disabled perhaps...? Am I missing documentation concerning this behavior?

Or is there a problem with 'extension handling' as the new file does not end in .toml and therefore should not be watched?

Watching an entire path is great for the situation I'm in, but the behavior is not at all clear :-s

**Comment by ldez** [comment]
The code you quote makes a difference between a directory and a file.
This code is the next version of Traefik (1.4.0).

I advise you to test with version 1.3.3.

**Comment by ldez** [comment]
In the previous versions (before the next version 1.4.0), we watch all files which contains `/etc/traefik/rules.toml`, e.g. `/etc/traefik/rules.toml-bak.20170724-132846` contains  `/etc/traefik/rules.toml` (+ `-bak.20170724-132846`).

workaround: add a prefix instead of a suffix.

This is  fixed in the next version (1.4.0).

**Comment by malc0mn** [comment]
Thanks for this clarification! Just make sure to mention this behaviour somewhere in the docs ;-)
